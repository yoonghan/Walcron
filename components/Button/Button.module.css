
  .container {
    padding: 1rem;
    position: relative;
    color: var(--button);
    background: var(--button-bg);
    transition-property: 'color', 'background';
    transition-duration: 0.15s;
    transition-timing-function: ease-in-out;
    color: #000;
    font-size: 1rem;
    text-align: center;
    border-radius: 0.5rem;
    font-weight: bold;
    cursor: pointer;
    border: 1px solid;
    margin: 0 1px 1px 0;
    user-select: none;
    white-space: nowrap;
    box-shadow: 0 2px 2px 0 rgba(22, 22, 22, 0.14),
      0 1px 5px 0 rgba(22, 22, 22, 0.12),
      0 3px 1px -2px rgba(22, 22, 22, 0.2);
  }
  .container:hover {
    background: var(--button);;
    color: var(--button-bg);;
  }
  .container:active {
    margin: 1px 0 0 1px;
  }
  .container.small {
    padding: 0.5rem;
    font-size: 0.7rem;
  }
  .container.invert {
    background: rgba(25, 25, 25, 1);
    color: #fff;
  }
  .container.invert:hover {
    color: rgba(25, 25, 25, 1);
    background: #fff;
  }