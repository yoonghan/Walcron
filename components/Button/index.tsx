`use strict`

import * as React from "react";
import Link from 'next/link';

export interface ButtonProps {
  href?: string;
  target?: string;
  onClickCallback?: () => void;
  small?: boolean;
}

const createButton = (children?:any, _onClickCallback?:() => void, target?: string, small?: boolean) => {
  return (
    <a className="btn-container" onClick={_onClickCallback?_onClickCallback:()=>{}} target={target}>
      {children}
      <style jsx>
        {`
          a {
            text-decoration: none;
          }
          .btn-container {
            padding: 1rem;
            position: relative;
            background: rgba(200,200,200,0.1);
            transition-property: color, background;
            transition-duration: .15s;
            transition-timing-function: ease-in-out;
            color: #FFF;
            font-size: ${small? '0.7rem': '1rem'};
            text-align: center;
            border-radius: 0.5rem;
            font-weight: bold;
            cursor: pointer;
            border: 1px solid;
            margin: 0 1px 1px 0;
            user-select: none;
            white-space: nowrap;
            box-shadow: 0 2px 2px 0 rgba(22,22,22,0.14), 0 1px 5px 0 rgba(22,22,22,0.12), 0 3px 1px -2px rgba(22,22,22,0.2)
          }
          .btn-container:hover {
            background: #FFF;
            color: #000;
          }
          .btn-container:active {
            margin: 1px 0 0 1px;
          }
        `}
      </style>
    </a>
  );
}

const Button: React.SFC<ButtonProps> = ({children, href, target, onClickCallback, small}) => {
  return (
    href?
      <Link href={href}>{createButton(children, undefined, target, small)}</Link>:
      createButton(children, onClickCallback, undefined, small)
  );
}

export default Button;
