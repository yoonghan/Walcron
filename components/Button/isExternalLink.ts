export const isExternalLink = (href: string) =>
  href && /^(http:\/\/)|(https:\/\/)|^\/\//.test(href)
