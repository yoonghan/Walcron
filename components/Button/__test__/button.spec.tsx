import * as React from 'react';
import { shallow } from 'enzyme';
import Button from "../index";

describe('Button', () => {
  it('renders normally', () => {
    const wrapper = shallow(<Button><div>Children</div></Button>);
    expect(wrapper.debug()).toMatchSnapshot();
  }),
  it('renders with small icon', () => {
    const wrapper = shallow(<Button href="sample" target="sample" small={true}/>);
    expect(wrapper.debug()).toMatchSnapshot();
  }),
  it('renders with inverted color', () => {
    const wrapper = shallow(<Button invert={true}/>);
    expect(wrapper.debug()).toMatchSnapshot();
  }),
  it('renders with click', () => {
    const wrapper = shallow(<Button onClickCallback={()=>{}}/>);
    expect(wrapper.debug()).toMatchSnapshot();
  })
});
