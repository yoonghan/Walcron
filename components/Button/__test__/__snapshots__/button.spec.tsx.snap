// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders normally 1`] = `
"<ForwardRef style={{...}}>
  <a onClick={[Function]} href={[undefined]} target={[undefined]} className=\\"jsx-252516482 btn-container\\">
    <div>
      Children
    </div>
    <JSXStyle id=\\"3810349888\\" dynamic={{...}}>
      a.__jsx-style-dynamic-selector{-webkit-text-decoration:none;text-decoration:none;}.btn-container.__jsx-style-dynamic-selector{padding:1rem;position:relative;background:rgba(200,200,200,0.1);-webkit-transition-property:color,background;transition-property:color,background;-webkit-transition-duration:.15s;transition-duration:.15s;-webkit-transition-timing-function:ease-in-out;transition-timing-function:ease-in-out;color:#FFF;font-size:1rem;text-align:center;border-radius:0.5rem;font-weight:bold;cursor:pointer;border:1px solid;margin:0 1px 1px 0;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;white-space:nowrap;box-shadow:0 2px 2px 0 rgba(22,22,22,0.14),0 1px 5px 0 rgba(22,22,22,0.12),0 3px 1px -2px rgba(22,22,22,0.2);}.btn-container.__jsx-style-dynamic-selector:hover{background:#FFF;color:#000;}.btn-container.__jsx-style-dynamic-selector:active{margin:1px 0 0 1px;}
    </JSXStyle>
  </a>
</ForwardRef>"
`;

exports[`Button renders with click 1`] = `
"<ForwardRef style={{...}}>
  <a onClick={[Function: onClickCallback]} href={[undefined]} target={[undefined]} className=\\"jsx-252516482 btn-container\\">
    <JSXStyle id=\\"3810349888\\" dynamic={{...}}>
      a.__jsx-style-dynamic-selector{-webkit-text-decoration:none;text-decoration:none;}.btn-container.__jsx-style-dynamic-selector{padding:1rem;position:relative;background:rgba(200,200,200,0.1);-webkit-transition-property:color,background;transition-property:color,background;-webkit-transition-duration:.15s;transition-duration:.15s;-webkit-transition-timing-function:ease-in-out;transition-timing-function:ease-in-out;color:#FFF;font-size:1rem;text-align:center;border-radius:0.5rem;font-weight:bold;cursor:pointer;border:1px solid;margin:0 1px 1px 0;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;white-space:nowrap;box-shadow:0 2px 2px 0 rgba(22,22,22,0.14),0 1px 5px 0 rgba(22,22,22,0.12),0 3px 1px -2px rgba(22,22,22,0.2);}.btn-container.__jsx-style-dynamic-selector:hover{background:#FFF;color:#000;}.btn-container.__jsx-style-dynamic-selector:active{margin:1px 0 0 1px;}
    </JSXStyle>
  </a>
</ForwardRef>"
`;

exports[`Button renders with inverted color 1`] = `
"<ForwardRef style={{...}}>
  <a onClick={[Function]} href={[undefined]} target={[undefined]} className=\\"jsx-2486039159 btn-container\\">
    <JSXStyle id=\\"3810349888\\" dynamic={{...}}>
      a.__jsx-style-dynamic-selector{-webkit-text-decoration:none;text-decoration:none;}.btn-container.__jsx-style-dynamic-selector{padding:1rem;position:relative;background:rgba(25,25,25, 1.0);-webkit-transition-property:color,background;transition-property:color,background;-webkit-transition-duration:.15s;transition-duration:.15s;-webkit-transition-timing-function:ease-in-out;transition-timing-function:ease-in-out;color:#FFF;font-size:1rem;text-align:center;border-radius:0.5rem;font-weight:bold;cursor:pointer;border:1px solid;margin:0 1px 1px 0;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;white-space:nowrap;box-shadow:0 2px 2px 0 rgba(22,22,22,0.14),0 1px 5px 0 rgba(22,22,22,0.12),0 3px 1px -2px rgba(22,22,22,0.2);}.btn-container.__jsx-style-dynamic-selector:hover{background:#FFF;color:#000;}.btn-container.__jsx-style-dynamic-selector:active{margin:1px 0 0 1px;}
    </JSXStyle>
  </a>
</ForwardRef>"
`;

exports[`Button renders with small icon 1`] = `
"<Link href=\\"sample\\">
  <ForwardRef style={{...}}>
    <a onClick={[Function]} href={[undefined]} target=\\"sample\\" className=\\"jsx-3422965696 btn-container\\">
      <JSXStyle id=\\"3810349888\\" dynamic={{...}}>
        a.__jsx-style-dynamic-selector{-webkit-text-decoration:none;text-decoration:none;}.btn-container.__jsx-style-dynamic-selector{padding:0.5rem;position:relative;background:rgba(200,200,200,0.1);-webkit-transition-property:color,background;transition-property:color,background;-webkit-transition-duration:.15s;transition-duration:.15s;-webkit-transition-timing-function:ease-in-out;transition-timing-function:ease-in-out;color:#FFF;font-size:0.7rem;text-align:center;border-radius:0.5rem;font-weight:bold;cursor:pointer;border:1px solid;margin:0 1px 1px 0;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;white-space:nowrap;box-shadow:0 2px 2px 0 rgba(22,22,22,0.14),0 1px 5px 0 rgba(22,22,22,0.12),0 3px 1px -2px rgba(22,22,22,0.2);}.btn-container.__jsx-style-dynamic-selector:hover{background:#FFF;color:#000;}.btn-container.__jsx-style-dynamic-selector:active{margin:1px 0 0 1px;}
      </JSXStyle>
    </a>
  </ForwardRef>
</Link>"
`;
